src/beta_inc.jl

```diff
@@ -0,0 +1,865 @@
+#ACM Transactions on Mathematical Software. Vol18, No3, September1992, Pages360-373
+#DLMF : https://dlmf.nist.gov/8.17#E1
+#Wikipedia : https://en.wikipedia.org/wiki/Beta_function#Incomplete_beta_function
+
+"""
+    beta_inc(a,b,x,y)
+
+Computes Incomplete Beta Function Ratios given by:
+```math
+I_{x}(a,b) = G(a,b) \\int_{0}^{x} t^{a-1}(1-t)^{b-1} dt,
+```
+and ``I_{y}(a,b) = 1.0 - I_{x}(a,b)``.
+given
+``B(a,b) = 1/G(a,b) = \\Gamma(a)\\Gamma(b)/\\Gamma(a+b)`` and y = 1.0 - x
+"""
+function beta_inc(a::Float64, b::Float64, x::Float64, y::Float64)
+    ans_x = 0.0
+    ans_y = 0.0
+    lambda = a - (a+b)*x
+    if a < 0.0 || b < 0.0
+        return error("a or b is negative")
+    elseif a == 0.0 && b == 0.0
+        return error("a and b are 0.0")
+    elseif x < 0.0 || x > 1.0
+        return error("x < 0 or x > 1")
+    elseif y < 0.0 || y > 1.0
+        return error("y < 0  or y > 1")
+    else
+        z = x + y - 1.0
+        if abs(z) > 3.0*eps()
+            return error("x + y != 1.0")         # ERROR HANDLING
+        end
+    end
+
+    if x == 0.0
+        return (0.0,1.0)
+    elseif y == 0.0
+        return (1.0,0.0)
+    elseif a == 0.0
+        return (1.0,0.0)
+    elseif b == 0.0
+        return (0.0,1.0)
+    end
+
+    epps = max(eps(), 1.0e-15)
+    if max(a,b) < 1.0E-3 * epps
+        return (b/(a+b), a/(a+b))
+    end
+    ind = false
+    a0 = a
+    b0 = b
+    x0 = x
+    y0 = y
+    if min(a0,b0) > 1.0
+        @goto l30
+    end
+#PROCEDURE FOR A0<=1 OR B0<=1
+    if x <= 0.5
+        @goto l10
+    end
+    ind = true
+    a0 = b
+    b0 = a
+    y0 = x
+    x0 = y
+
+    @label l10
+     if b0 < min(epps, epps*a0)
+        @goto l80
+     elseif a0 < min(epps, epps*b0)
+        @goto l90
+     elseif max(a0,b0) > 1.0
+        if b0 <= 1.0 || (x0*b0)^a0 <= 0.7
+            @goto l100
+        elseif x0 >= 0.3
+            @goto l110
+        elseif x0 >= 0.1
+            if b0 > 15.0
+                @goto l131
+            else
+                n = 20
+                @goto l130
+            end
+        end
+     elseif a0 >= min(0.2, b0) || x0^a0 <= 0.9
+        @goto l100
+     elseif x0 >= 0.3
+        @goto l110
+     else
+        n = 20
+        @goto l130
+     end
+
+#PROCEDURE FOR A0>1 AND B0>1
+    @label l30
+     if a > b
+        lambda = (a+b)*y - b
+     else
+        lambda = a - (a+b)*x
+     end
+     if lambda >= 0.0
+        @goto l40
+     end
+    
+    ind = true
+    a0 = b
+    b0 = a
+    x0 = y
+    y0 = x
+
+    lambda = abs(lambda)
+    @label l40
+     if b0 < 40.0 && b0*x0 <= 0.7
+
+        @goto l100
+     elseif b0 < 40.0
+        @goto l140
+     elseif a0 > b0
+        if b0 <= 100.0 || lambda > 0.03*b0
+            @goto l120
+        else
+            @goto l180
+        end
+     elseif a0 <= 100.0 || lambda > 0.03*a0
+        @goto l120
+     else
+        @goto l180
+     end
+
+#EVALUATION OF ALGOS FOR PROPER SUB-DOMAINS
+
+    @label l80
```

Comment:
from here down: for each one of these that can only be reached from one place, move to the source `@goto`.